include/EVT/io/SPI.hpp

```diff
@@ -0,0 +1,143 @@
+#ifndef _EVT_SPI_
+#define _EVT_SPI_
+
+#include "GPIO.hpp"
+#include <stdint.h>
+
+#define SPI_MSB_FIRST 1
+#define SPI_LSB_FIRST 0
+
+#define SPI_MODE0 0x01
+#define SPI_MODE1 0x02
+#define SPI_MODE2 0x04
+#define SPI_MODE3 0x08
+
+#define SPI_SPEED_4MHZ 4000000
+#define SPI_SPEED_2MHZ 2000000
+#define SPI_SPEED_1MHZ 1000000
+#define SPI_SPEED_500KHZ 500000
+#define SPI_SPEED_250KHZ 250000
+#define SPI_SPEED_125KHZ 125000
+#define SPI_SPEED_62KHZ 62500
+#define SPI_SPEED_31KHZ 31250
+
+namespace EVT::core::IO {
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+enum class Pin;
+
+class SPI {
+public:
+    /**
+         * Make an instance of an I2C interface that will use the given pins
+         * for clock and data lines.
+         *
+         * @param sckPin[in] The SPI clock pin
+         * @param mosiPin[in] The mosi pin (data out)
+         * @param misoPin[in] The miso pin (data in)[optional]
+         */
+    SPI(GPIO** CSPins, uint8_t pinLength, Pin sckPin, Pin mosiPin, Pin misoPin);
+
+    SPI(GPIO** CSPins, uint8_t pinLength, Pin sckPin, Pin mosiPin);
+
+    /**
+         * begin a device transmission. Call before each set of read and write interactions.
```

Comment:
Note: clang-format doesn't format comments, so that is why the comments now looked over tabbed. Unfortunately we do have to change that manually.

Reply:
yeah, I wasn't expecting the code to get reformatted in the pull request.